### PR TITLE
initial destination door combobox, unselectDoor when change edit mode

### PR DESCRIPTION
**Pull request by shinespeciall**

### CODE REVIEW

File: Dialog/DoorConfigDialog.cpp [hunk 1]
```diff
@@ -106,3 +126,42 @@ void DoorConfigDialog::RenderGraphicsView_DestinationDoor(int doorIDinRoom)
...
+void DoorConfigDialog::on_ComboBox_DoorDestinationPicker_currentIndexChanged(int index)
+{
+    delete tmpDestinationRoom;
+    tmpDestinationRoom = new LevelComponents::Room(_currentLevel->GetRooms()[_currentLevel->GetDoors()[index]->GetRoomID()]);
+    tmpDestinationRoom->SetDoors(_currentLevel->GetRoomDoors((unsigned int) _currentLevel->GetDoors()[index]->GetRoomID()));
+    _currentLevel->GetDoors()[index]->SetDestinationDoor(_currentLevel->GetDoors()[index]); //TODO: this does not work.
+    RenderGraphicsView_DestinationDoor(tmpDestinationRoom->GetLocalDoorID(index));
+}
+
+void DoorConfigDialog::on_SpinBox_DoorX_valueChanged(int arg1)
```
Comment (Goldensunboy) [inline Dialog/DoorConfigDialog.cpp]: Documentation

File: Dialog/DoorConfigDialog.cpp [hunk 2]
```diff
@@ -106,3 +126,42 @@ void DoorConfigDialog::RenderGraphicsView_DestinationDoor(int doorIDinRoom)
...
+    tparam.tileX = tparam.tileY = 0; tparam.tileID = (unsigned short) 0;
+    tparam.SelectedDoorID = (unsigned int) DoorID; //ID in Room
+    tparam.mode.editMode = Ui::DoorEditMode;
+    tparam.mode.entitiesEnabled = tparam.mode.cameraAreasEnabled = false;
+    QGraphicsScene *scene = tmpCurrentRoom->RenderGraphicsScene(ui->GraphicsView_Preview->scene(), &tparam);
+    ui->GraphicsView_DestinationDoor->setScene(scene);
+    ui->GraphicsView_DestinationDoor->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+}
+
+void DoorConfigDialog::on_ComboBox_DoorDestinationPicker_currentIndexChanged(int index)
```
Comment (Goldensunboy) [inline Dialog/DoorConfigDialog.cpp]: Documentation

File: WL4EditorWindow.cpp [hunk 3]
```diff
@@ -178,7 +178,12 @@ void WL4EditorWindow::SetEditModeDockWidgetLayerEditability()
 
 bool *WL4EditorWindow::GetLayersVisibilityArray()
```
Comment (Goldensunboy) [inline WL4EditorWindow.cpp]: Documentation

File: WL4EditorWindow.cpp [hunk 4]
```diff
@@ -178,7 +178,12 @@ void WL4EditorWindow::SetEditModeDockWidgetLayerEditability()
...
 {
-   return EditModeWidget->GetLayersVisibilityArray();
+    return EditModeWidget->GetLayersVisibilityArray();
+}
+
+void WL4EditorWindow::Graphicsview_UnselectDoor()
```
Comment (Goldensunboy) [inline WL4EditorWindow.cpp]: Documentation

File: DockWidget/EditModeDockWidget.cpp [hunk 5]
```diff
@@ -220,4 +220,10 @@ void EditModeDockWidget::on_CheckBox_AlphaView_stateChanged(int arg1)
     singleton->RenderScreenVisibilityChange(); // TODO this should probably be a full re-render
 }
 
-
+void EditModeDockWidget::on_RadioButton_DoorMode_toggled(bool checked)
```
[the remark this answers is not in the document]
Comment (shinespeciall) [reply DockWidget/EditModeDockWidget.cpp]: Done.

File: LevelComponents/EntitySet.cpp [hunk 6]
```diff
@@ -128,6 +130,34 @@ namespace LevelComponents
         // TODOs: set other entity informations
     }
 
+    int EntitySet::EntitySetFromEntityID(int entityglobalId)
```
[the remark this answers is not in the document]
Comment (shinespeciall) [reply LevelComponents/EntitySet.cpp]: Done.

File: LevelComponents/EntitySet.cpp [hunk 7]
```diff
@@ -128,6 +130,34 @@ namespace LevelComponents
...
+                {
+                    return j;
+                }
+                i++;
+            }
+        }
+        return 0; //TODO: Error handling
+    }
+
+    int EntitySet::GetEntityFirstActionFrameSetPtr(int entityglobalId)
```
[the remark this answers is not in the document]
Comment (shinespeciall) [reply LevelComponents/EntitySet.cpp]: Done.

File: Dialog/DoorConfigDialog.cpp (hunk 1 above)
Comment (shinespeciall) [reply Dialog/DoorConfigDialog.cpp]: Done.

File: WL4EditorWindow.cpp (hunk 3 above)
Comment (shinespeciall) [reply WL4EditorWindow.cpp]: Done.

File: WL4EditorWindow.cpp (hunk 4 above)
Comment (shinespeciall) [reply WL4EditorWindow.cpp]: Done.

**Pull request merged**